server/svix-server/src/v1/endpoints/endpoint/crud.rs

```diff
@@ -189,6 +201,17 @@ pub(super) async fn update_endpoint(
         validate_event_types(db, event_types_ids, &app.org_id).await?;
     }
     validate_endpoint_url(&data.url, cfg.endpoint_https_only)?;
+    
+    // Validate CEL expression if provided
+    if let Some(ref filter) = data.filter {
+        if let Err(e) = validate_cel_expression(filter) {
+            return Err(HttpError::unprocessable_entity(vec![ValidationErrorItem {
+                loc: vec!["filter".to_string()],
+                msg: format!("Invalid CEL expression: {}", e),
+                ty: "value_error".to_string(),
+            }]).into());
```

Comment:
This is worth extracting into a function and calling it like `create_invalid_cel_expression_error(e)`.